### DISCUSSION TITLE
[Product] 목표 금액 설정에 대한 논의

### DISCUSSION THREAD

**Discussion by psychology50** [CollaBu/pennyway-was#5] [Q&A] [answered]
현재 Pennyway 서비스에는 다음과 같은 기능이 있습니다.

<div align="center">
   <img src="https://github.com/CollaBu/pennyway-was/assets/96044622/36f04a03-1539-4e5a-b0dd-38ed093144f3" width="600px" />
</div>

<div align="center">
   <img src="https://github.com/CollaBu/pennyway-was/assets/96044622/74293055-112b-45aa-b861-d6f9f51fd5af" width="300px"/>
</div>

- 최초 회원가입한 사용자는 로그인 후 목표 금액을 설정할 수 있는 화면을 출력한다.
- 목표 금액은 매월마다 구분하여 관리하며, 목표 달성 여부를 판단한다.

여기서 논점은 신규 가입자가 아닌 "`기존 사용자`의 다음 달 목표 금액을 어떻게 설정하는가?"입니다.

1. 매월 서비스에 접속하는 유저는 목표 금액을 설정하도록 최초 로그인 시와 동일한 스크린을 보여준다.
   - 사용자가 매달 목표 금액을 다르게 설정할 수 있음을 명시적으로 확인시켜 줄 수 있다.
   - 사용자가 목표 금액 설정 스크린까지 이동하는 절차 없이, 앱을 실행했을 때 바로 설정 가능하다.
   - 빈도 수가 낮기 때문에 사용자 경험을 떨어트릴 정도는 아니라고 생각하지만, iOS 팀에서 개발 난이도가 높아질 우려가 있다.

2. 달이 바뀌면 서버에서 지난 달의 목표 금액을 동일하게 설정해주고, 사용자가 변경하도록 한다. 
   - 사용자가 매달 목표 설정 스크린을 보지 않아도 된다.
   - 어차피 서버에서는 매달 사용자의 목표 설정 레코드를 생성해주어야 하므로, 추가적인 api가 생기지 않는다.

3. 다른 의견

개인적으로는 `(1)`의 방식을 채택하고자 하는데, 여러분들의 의견을 먼저 들어보는 것이 좋을 것 같아 Discussions에 남깁니다.  
해당 논의는 `3월 15일 (금)` 스프린트 회고 회의 후 시간이 남으면 투표할 계획입니다.

**Comment by yanni13** [comment]
저는 `2. 달이 바뀌면 서버에서 지난 달의 목표 금액을 동일하게 설정해주고, 사용자가 변경하도록 한다. `를 채택하는게 좋을거 같다는 의견입니다.

개인적으로 사용자가 한 번 목표 금액을 설정한 이후 고정지출이 더 나가거나, 수입이 더 들어오지 않는 한 목표 금액이 몇개월간 크게 변동 될 일은 없다고 생각합니다. 따라서 사용자가 매달 목표 금액을 설정하도록 하는 것 보단 필요시에 변경하도록 하는게 좋을 거 같다는 생각이 드네요.

**Reply by psychology50** [reply]
저도 이 부분 때문에 고려했던 시나리오긴 합니다. 좋은 의견 감사합니다.

**Comment by BangDori** [comment]
3. 다른 의견

저는 2번 의견에서 추가로 다음과 같은 메시지(메시지는 가공해야 할 듯 합니당)를 버튼으로 표시하는 것도 괜찮을 거 같습니다.

- 만약 금액이 **3만원 만큼 초과**되었다면 (기존 30만) -> 이번 달 목표 금액을 **33만원**으로 설정하시겠습니까?
- 만약 금액이 **3만원 만큼 절약**되었다면 (기존 30만) -> 이번 달 목표 금액을 **27만원**으로 설정하시겠습니까?

**Reply by psychology50** [reply]
저희는 지출 욕구를 억제시켜야 하는 서비스기 때문에 절약했을 때 줄인 금액을 제시하는 건 좋아보이지만, 초과 했을 때 한계 금액을 늘리는 걸 제시하는 건 위험할 수 있다고 생각합니다.  
병준님 의견을 살려보자면 절약했을 때는 절약한 만큼 목표 금액을 제시하고, 초과했을 때는 유지하는 방식이어야 할 거 같습니다.

**Reply by BangDori** [reply]
재서님이 말씀하신 것처럼 초과했을 때 한계 금액을 늘리는 것은 서비스의 의도와 부합하지 않다고 볼 수 있을 것 같습니다.

하지만 서비스를 이용하는 사용자가 **1~2달 만에 적절한 목표 금액을 설정하기에는 무리**가 있을 수도 있다는 생각이 듭니다. 그래서 제가 생각하기에는 적절한 목표 금액을 설정할 수 있도록 안내해주는 것 또한 서비스가 제공해줄 수 있는 기능이라고 생각합니다.

혹시 어떻게 생각하시나요?

**Reply by psychology50** [reply]
그럼 이제 2가지로 다시 경우가 갈리게 될 것 같은데.

1. 첫 두 달 동안은 초과/절약 금액만큼 목표 금액을 조장할 것인지 여부를 확인한다.
2. 지속적으로 초과/절약 금액만큼 조정할 것인지 확인한다.

`1`의 방식은 합리적이지만, 필요 이상으로 서비스 기능이 복잡해질 수 있습니다.  
추가 UI가 필요하며, 기능이 더 많아지고 그 이후에는 어떻게 표시할 것인지에 대해 결국 정책 논의가 원점으로 돌아옵니다.

두 번째 방법은 처음 말씀드린 것처럼 서비스 성격의 문제입니다.  
서비스가 초기 사용자들에게 도움을 줄 수는 있으나, 그것이 장기 유저에게까지 지속적으로 의사를 묻게 된다면 이는 정책 실패라고 생각합니다.

**Reply by psychology50** [reply]
Pennyway는 목표 금액보다 절약한 사람에겐 보상을 주고, 초과한 사람에겐 패널티를 주어야 합니다.  
그래야 사용자들이 보상을 얻기 위해 더 절약하게 될 테니까요.  

하지만 절약을 하나, 초과를 하나 서비스가 언제나 친절하게 "다음엔 목표 기준치를 다르게 잡아보실래요?"라고 묻는다면, 소비 억제 요구가 반감되진 않을 것이라는 우려가 있습니다.

**Reply by BangDori** [reply]
음 재서님의 의견을 종합해서 고려해보니, 2번 선택지에는 확실히 무리가 있어 보입니다.

근데 1번 선택지에는 서비스 기능이 복잡해질 우려도 있고, 현재로서는 확실한 답안이 서지 않는 것도 사실이지만 서비스가 사용자에게 한층 더 가까워질 수 있다는 생각이 듭니다.

우선 `version 1.0.0`에서는 재서님이 제시해주신 2번 의견을 채택하여 서비스를 제공해주고, 이후에 다시 한 번 discussion을 통해 이야기를 나눠보는 것도 좋을 것 같습니다.

**Comment by psychology50** [comment]
한 가지 추가 의견.  

기존 목표 금액을 동일하게 가져가면 문제가 될 수 있는 게, 사용자가 3월에 서비스를 사용하다가, 4월에 사용하지 않고, 5월에 다시 복귀했다고 가정합시다.  

`2` 방식은 사용자가 마치 4월에 지난 달 목표 금액 만큼 절약했다는 것처럼 보이게 할 수도 있습니다.  
(3월 목표금액인 10,000원을 그대로 4월 목표 금액으로 설정했는데, 서비스를 사용하지 않고 있었으니 지출은 0원이 되므로)  

따라서, 다음 달이 되면 목표 금액은 미설정 값으로 할당하되 사용자가 로그인 하면 "지난 달 목표 금액으로 설정하시겠습니까?" 라는 알림을 띄워서 간단하게 설정하는 방법도 존재합니다.

**Reply by asn6878** [reply]
재서님이 말씀해주신 장기간 서비스 사용의 부재가 발생하는 시나리오까지 고려를 하자면, 알림 텍스트는
"지난 달 목표 금액으로 설정하시겠습니까?" 보다는
"가장 최근 목표 금액으로 설정하시겠어요?" 가 적절해 보인다는 생각 입니다!

**Reply by psychology50** [reply]
좋은 멘트라고 생각합니다. 의견 감사합니다.

**Comment by jinlee1703** [comment, marked as answer]
- (바로 위에 재서님이 제시해주셨던) 다음 달이 되면 목표 금액은 미설정 값으로 할당하되 사용자가 로그인 하면 "지난 달 목표 금액으로 설정하시겠습니까?" 라는 알림 띄우기 => `아니오` 클릭 시 매월 서비스에 접속하는 유저는 목표 금액을 설정하도록 최초 로그인 시와 동일한 스크린을 보여주기

이 두 가지가 `목표액과 차익 비교`에 더하여 `매월 본인이 스스로 목표를 설정`함으로써 성취감을 줄 수 있지 않을까 생각합니다...!

**Reply by psychology50** [reply]
아니면 모두 동일하게 스크린을 띄우되, 기존 사용자는 지난 달에 입력한 목표 금액을 미리 입력폼에 넣어주는 방법도 있을 거 같긴 하네요.  
그래도 쓸 사람은 확인 버튼만 누르면 되니까요.

**Comment by 2weeksone** [comment]
디자인팀 회의에서 재서님과 상의 후 나온 의견 전달드립니다.

현재 해당 알림의 선택지에서는 **유저가 결정을 하지 않는 선택지**가 없습니다.
서비스 이용 중 결정을 하지 않는 선택지 (ex. 다음에 하기, 취소, 무시하기)가 없다면, 유저가 귀찮음을 느낄 가능성이 있습니다.

따라서 ‘가장 최근 목표 금액으로 설정할까요?’라는 알림에
선택지를 ‘설정하기'-’무시하기' 두 가지로 하여

**‘설정하기’ 선택시**, 가장 최근 목표 금액으로 설정.
**‘무시하기’ 선택시**, 유저가 지출관리 페이지에서 원할 때 설정. (화면 이동 없이 알림창 닫기)

하는 것이 적절하다고 생각됩니다.

**Reply by psychology50** [reply]
월요일에 해당 안건을 가지고 다시 한 번 이야기해보도록 하겠습니다.  
의견 감사드립니다.
